File: addon-sdk/source/lib/sdk/test/memory.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
'use strict';

const { Cu } = require("chrome");
const memory = require('../deprecated/memory');
const { defer } = require('../core/promise');

function gc() {
  let { promise, resolve } = defer();

  Cu.forceGC();
  memory.gc();

  Cu.schedulePreciseGC(_ => resolve());

  return promise;
}
exports.gc = gc;

File: mobile/android/geckoview/src/androidTest/java/org/mozilla/geckoview/test/rdp/Memory.java
/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.geckoview.test.rdp;

/**
 * Provide access to the memory API.
 */
public final class Memory extends Actor {
    /* package */ Memory(final RDPConnection connection, final String name) {
        super(connection, name);
    }

    public void forceCycleCollection() {
        sendPacket("{\"type\":\"forceCycleCollection\"}", JSON_PARSER).get();
    }

    public void forceGarbageCollection() {
        sendPacket("{\"type\":\"forceGarbageCollection\"}", JSON_PARSER).get();
    }
}
